x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -161,13 +161,14 @@ public LocalExecutionPlanner(
     /**
      * turn the given plan into a list of drivers to execute
      */
-    public LocalExecutionPlan plan(PhysicalPlan localPhysicalPlan) {
+    public LocalExecutionPlan plan(FoldContext foldCtx, PhysicalPlan localPhysicalPlan) {
```

Comment:
Why do you need here to pass as parameter a `FoldContext`?
From what I can tell the constructor of `LocalExecutionPlanner` already has the needed pragma to build the folding context.

Also, it's unfortunate that we must build the `PhysicalOperationProviders` (which also takes a folding context) outside the `LocalExecutionPlanner` (for testing purposes that is) otherwise `PhysicalOperationProviders` could have been built as part of the `LocalExecutionPlanner` build and would have gotten the same folding context as the surrounding class.

Reply:
I had wanted to use a single context so we'd share the same limit across the whole process. I *think* at some point we'll no longer need `FoldContext` at the physical level at all *eventually* so maybe it's mute?

Reply:
I'm going to double check exactly how much sharing I get here.

Reply:
Right. We share with the the `LocalLogicalPlanOptimizer` and `LocalPhysicalPlanOptimizer`.